app/controllers/microposts_controller.rb

```diff
@@ -0,0 +1,31 @@
+class MicropostsController < ApplicationController
+  before_action :logged_in_user, only: [:create, :destroy]
+  before_action :correct_user, only: :destroy
+
+  def create
+    @micropost = current_user.microposts.build micropost_params
+    if @micropost.save
+      flash[:success] = t "micropost_success"
+      redirect_to root_url
+    else
+      @feed_items = current_user.feed.paginate page: params[:page]
+      render "static_pages/home"
+    end
+  end
+
+  def destroy
+    @micropost.destroy
+    flash[:success] = t "micropost_deleted"
+    redirect_to request.referrer || root_url
+  end
+
+  private
+  def micropost_params
+    params.require(:micropost).permit :content, :picture
+  end
+
+  def correct_user
```

Comment:
Nếu k đưa **:picture** vào permit thì khi post sẽ k thể hiện ảnh, nếu k đưa **:content** và post thì k thể đăng bài vì cho dù mình có viết thì phương thức đó cũng là k được phép, dữ liệu đưa vào rỗng sẽ trả về kết quả: **"content can not be blank"** do mình đang sử dụng validate  **presence: true** trong micropost.rb